### PR TITLE
Spawn enemies in random location around player

### PR DESCRIPTION
**Pull request by CooperativeDev**
created MobSpawner class which spawns mobs at random points in a circle around the player 

currently spawns a new mob every 2 seconds

### CODE REVIEW

File: Assets/Scripts/Mob/MobSpawner.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
+using UnityEngine;
+using System;
+using Random = System.Random;
+
+namespace Mob
+{
+    public class MobSpawner : MonoBehaviour
+    {
+        [SerializeField] private GameObject mobPrefab;
+        private Vector3 _floorDimensions;
```
Comment (jamilfaisal) [inline Assets/Scripts/Mob/MobSpawner.cs]: Attribute not required. Call transform.localScale.x on line 22

File: Assets/Scripts/Mob/MobSpawner.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        private int _radius;
+        private double _randomAngleRadians;
```
Comment (jamilfaisal) [inline Assets/Scripts/Mob/MobSpawner.cs]: Convert this to a local variable

File: Assets/Scripts/Mob/MobSpawner.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        private int _xCoordinate;
```
Comment (jamilfaisal) [inline Assets/Scripts/Mob/MobSpawner.cs]: Local variable

File: Assets/Scripts/Mob/MobSpawner.cs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        private int _zCoordinate;
```
Comment (jamilfaisal) [inline Assets/Scripts/Mob/MobSpawner.cs]: Local variable

File: Assets/Scripts/Mob/MobSpawner.cs [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+            if (SpawnTimer())
+            {
+                SpawnMob();
+            }
+        }
+
+        private void SpawnMob()
+        {
+            MobSpawnCoordinates();
+            Instantiate(mobPrefab, new Vector3(_xCoordinate, 10, _zCoordinate), Quaternion.identity);
```
Comment (jamilfaisal) [inline Assets/Scripts/Mob/MobSpawner.cs]: private void SpawnMob() 
{ 
    Vector3 mobSpawnCoordinates = CalculateMobSpawnCoordinates();
    Instantiate(mobPrefab, mobSpawnCoordinates, Quaternion.identity);
}

private Vector3 CalculateMobSpawnCoordinates() 
{
      double randomAngleRadians = _randomNumber.Next(360)*(Math.PI/180);
      int xCoordinate = (int)(_radius * Math.Sin(_randomAngleRadians));
      int zCoordinate = (int)(_radius * Math.Cos(_randomAngleRadians));
      int yCoordinate = transform.position.y + 10;
      return new Vector3(xCoordinate, yCoordinate, zCoordinate);
  }

**Review by jamilfaisal — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Mob/MobSpawner.cs (hunk 1 above)
Comment (CooperativeDev) [reply Assets/Scripts/Mob/MobSpawner.cs]: cool

File: Assets/Scripts/Mob/MobSpawner.cs (hunk 2 above)
Comment (CooperativeDev) [reply Assets/Scripts/Mob/MobSpawner.cs]: i am just a salesman.
ok

File: Assets/Scripts/Mob/MobSpawner.cs (hunk 3 above)
Comment (CooperativeDev) [reply Assets/Scripts/Mob/MobSpawner.cs]: cool
i am just a salesman.
ok
thanks

File: Assets/Scripts/Mob/MobSpawner.cs (hunk 4 above)
Comment (CooperativeDev) [reply Assets/Scripts/Mob/MobSpawner.cs]: cool
i am just a salesman.
ok
thanks

File: Assets/Scripts/Mob/MobSpawner.cs (hunk 5 above)
Comment (CooperativeDev) [reply Assets/Scripts/Mob/MobSpawner.cs]: cool
i am just a salesman.
ok
thanks

**Review by jamilfaisal — APPROVED** [review]
LGTM!

**Pull request merged**